### DISCUSSION TITLE
Saving drawings

### DISCUSSION THREAD

**Discussion by aliveoceans** [klinecharts/KLineChart#486] [General]
Is there a way to save drawings so that they can be retrieved after revisiting the chart?

**Comment by kwanso-saif** [comment]
So in registerOverlay store the overlay that you just created:
  
      onDrawEnd: function (event) {
        localStorage.setItem("MYEVENT", JSON.stringify(event.overlay))
        return true;
      },
      
  And in UseLayoutEffect one can retireve the already made overlays:
  
        if (localStorage.getItem("MYEVENT")) {
        console.log(JSON.parse(localStorage.getItem("MYEVENT") || ""))
        chart.createOverlay(JSON.parse(localStorage.getItem("MYEVENT") || ""))
      }

**Reply by ct152098** [reply]
What if it's a built-in overlay? How can I monitor the method of drawing the end of a line?
